@navikt/core/react/src/button/Button.tsx

```diff
@@ -82,29 +81,14 @@ export const Button: OverridableComponent<ButtonProps, HTMLButtonElement> =
       ref,
     ) => {
       const buttonRef = useRef<HTMLButtonElement | null>(null);
-      const [widthOverride, setWidthOverride] = useState<number>();
 
       const mergedRef = useMergeRefs(buttonRef, ref);
 
-      useClientLayoutEffect(() => {
-        if (loading) {
-          const requestID = window.requestAnimationFrame(() => {
-            setWidthOverride(
-              buttonRef?.current?.getBoundingClientRect()?.width,
-            );
-          });
-          return () => {
-            setWidthOverride(undefined);
-            cancelAnimationFrame(requestID);
-          };
-        }
-      }, [loading, children]);
-
       const filterProps: React.ButtonHTMLAttributes<HTMLButtonElement> =
-        disabled ?? widthOverride ? omit(rest, ["href"]) : rest;
+        disabled ?? loading ? omit(rest, ["href"]) : rest;
```

Comment:
This line seems like it never worked as intended. Shouldn't it be as follows?

```suggestion
        (disabled || loading) ? omit(rest, ["href"]) : rest;
```

Reply:
Nice catch 🐛

Reply:
Indeed scary to have lots of boolean logic without parentheses 😅 (I try to always add too many as I never trust myself there, and in some cases, the linter will remove the redundant ones).

Reply:
I think the reason TypeScript doesn't catch this is because none of the props are typed after going through `forwardRef`. Tried adding the type `<HTMLButtonElement, ButtonProps>` to `forwardRef`, but that did not accept the `as`-prop. Any ideas?

Reply:
`OverridableComponent` creates some challenges with types inside the component. I don't really think there's a way to get "perfect" types as long as we accept any tag/component in `as`. You _could_ add this on line 81: `}: ButtonProps & { as?: React.ElementType },` but then you get issues when referring to `href`. I guess you could override that instance with `as any` though and still end up with improved types overall.